core/controllers/classroom.py

```diff
@@ -101,3 +113,93 @@ class DefaultClassroomRedirectPage(base.BaseHandler):
     def get(self):
         """Handles GET requests."""
         self.redirect('/learn/%s' % constants.DEFAULT_CLASSROOM_URL_FRAGMENT)
+
+
+class ClassroomAdminDataHandler(base.BaseHandler):
+    """Fetches relevant data for the classroom admin page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {'GET': {}}
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        """Handles GET requests."""
+        classroom_id_to_classroom_name = (
+            classroom_config_services.get_classroom_id_to_classroom_name_dict())
+
+        self.values.update({
+            'classroom_id_to_classroom_name': classroom_id_to_classroom_name
+        })
+        self.render_json(self.values)
+
+
+class NewClassroomIdHandler(base.BaseHandler):
+    """Creates a new classroom ID."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {'GET': {}}
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        """Handles GET requests."""
+        self.values.update({
+            'classroom_id': classroom_config_services.get_new_classroom_id()
+        })
+        self.render_json(self.values)
+
+
+class ClassroomHandler(base.BaseHandler):
+    """Edits classroom data."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'classroom_id': {
+            'schema': SCHEMA_FOR_CLASSROOM_ID
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'classroom_dict': {
+                'schema': {
+                    'type': 'object_dict',
+                    'object_class': classroom_config_domain.Classroom
+                }
+            }
+        },
+        'DELETE': {}
+    }
+
+    @acl_decorators.can_access_admin_page
+    def get(self, classroom_id):
+        """Handles GET requests."""
+        classroom = classroom_config_services.get_classroom_by_id(
+            classroom_id, strict=False)
+        if classroom is None:
+            raise self.PageNotFoundException(
+                'The classroom with the given id or url doesn\'t exist.')
+
+        self.values.update({
+            'classroom_dict': classroom.to_dict()
+        })
+        self.render_json(self.values)
+
+    @acl_decorators.can_access_admin_page
+    def put(self, classroom_id):
+        """Updates properties of a given classroom."""
+        classroom = self.normalized_payload.get('classroom_dict')
+        assert classroom_id, classroom.classroom_id
```

Comment:
This seems odd. Why can't this be validated by schema?

Or are you trying to do an equality check? If so, the pattern should be "if condition does not hold, raise exception".

Reply:
Updated